drawBot/drawBotDrawingTools.py

```diff
@@ -496,6 +496,26 @@ def restore(self):
         self._requiresNewFirstPage = True
         self._addInstruction("restore")
 
+    def savedState(self):
+        """
+        Save and restore the current state in a `with` statement.
+
+        .. downloadcode:: savedState.py
+
+            # use with statement
+            # this will be wrapped around a `save` and `restore`
+            with savedState():
+                # set a color
+                fill(1, 0, 0)
```

Comment:
I would love a transformation in the example, too, like rotate and/or translate. I hardly ever use (or need) a saved graphics state for paint styles, but all the time for transformations.